apps/election-manager/src/screens/SmartcardsScreen.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useContext, useState } from 'react'
+
+import { ButtonEventFunction } from '../config/types'
+import AppContext from '../contexts/AppContext'
+
+import NavigationScreen from '../components/NavigationScreen'
+import Prose from '../components/Prose'
+import Button from '../components/Button'
+import Modal from '../components/Modal'
+import ProgressEllipsis from '../components/ProgressEllipsis'
+
+const DefinitionScreen: React.FC = () => {
+  const { electionDefinition } = useContext(AppContext)
+  const { election, electionHash } = electionDefinition!
+
+  const [isProgrammingCard, setIsProgrammingCard] = useState(false)
+
+  const programCard: ButtonEventFunction = async (event) => {
+    const { id } = event.currentTarget.dataset
+    setIsProgrammingCard(true)
+
+    if (id === 'override') {
+      await fetch('/card/write_protect_override', {
+        method: 'post',
+      })
+      window.setTimeout(() => {
+        setIsProgrammingCard(false)
+      }, 1000)
+      return
+    }
+
+    const electionJSON = JSON.stringify(election)
+    const shortValue = JSON.stringify({
+      t: id,
+      h: electionHash,
+    })
+
+    const formData = new FormData()
+
+    switch (id) {
+      case 'pollworker':
+        await fetch('/card/write', {
+          method: 'post',
+          body: shortValue,
+        })
+        break
+      case 'admin':
+        formData.append('short_value', shortValue)
+        formData.append('long_value', electionJSON)
+        await fetch('/card/write_short_and_long', {
+          method: 'post',
+          body: formData,
+        })
+        break
+      default:
+        break
+    }
+
+    setIsProgrammingCard(false)
+  }
+
+  return (
+    <React.Fragment>
+      <NavigationScreen mainChildFlex>
+        <Prose maxWidth={false}>
+          <h1>Smartcards</h1>
+          <p>
+            Insert a card into the reader and then select the type of card to
+            create.
+          </p>
+          <p>
+            <Button onPress={programCard} data-id="admin">
+              Encode Admin Card
+            </Button>{' '}
+            <Button onPress={programCard} data-id="pollworker">
+              Encode Poll Worker Card
+            </Button>
+          </p>
+          <p>
+            You will first need to override write protection before
+            re-programming an existing Admin card.
+          </p>
+          <p>
+            <Button small onPress={programCard} data-id="override">
+              Override Write Protection
+            </Button>
+          </p>
+          {isProgrammingCard && <p>Is programming card…</p>}
+        </Prose>
+      </NavigationScreen>
+      {isProgrammingCard && (
+        <Modal
+          content={
+            <Prose textCenter>
+              <ProgressEllipsis>Programming card</ProgressEllipsis>
```

Comment:
Shouldn't we use the Loading component which wraps this?

Reply:
Yep, you're right. Will do. Not sure why I forgot about it. This thing needs a refactor. Might also be worth considering some other more obvious indeterminate progress indicator.

> Do we expect people to understand what write protection means?

I'd argue that we should get rid of write protection for simplicity, but perhaps there is a use case @benadida can add to clarify. Perhaps if the card being programmed is not blank, we add an additional confirmation step to ensure that they wish to overwrite the data on the card.